packages/paste-website/src/components/tokens-list/index.tsx

```diff
@@ -113,6 +116,28 @@ export const TokensList: React.FC = () => {
     trackTokenFilterString(filterString);
   }, [filterString, tokens]);
 
+  // Intersection observer for the Token Filter UI. The Tokens Filter gets a shadow
+  // as the user scrolls past its inherent position, and this intersection observer
+  // checks against a 'canary' element to determine when the shadow should be applied
+  React.useEffect(() => {
+    const intObserver = new IntersectionObserver(
+      (entries) => {
+        const shadowState = entries[0].intersectionRatio;
+        setShadowOpacity(1 - shadowState);
+      },
+      {
+        root: document.querySelector('#styled-site-body'),
+        rootMargin: '0px',
+        threshold: [1, 0.8, 0.6, 0.4, 0.2, 0],
+      }
+    );
+
+    const tokensFilter = document.querySelector('#filter-canary');
+    if (tokensFilter) {
+      intObserver.observe(tokensFilter);
+    }
```

Comment:
suggestion: you might need to add a clean up effect that disconnects the observer by doing soemthing like

```
return () => {
  intObserver.disconnect()
}
```